File: valid-sudoku.cs
// https://leetcode.com/problems/valid-sudoku/

// Congrats!!

public class Solution {
    public bool IsValidSudoku(char[][] board) {
        bool[,] rows = new bool[9,9];
        bool[,] cols = new bool[9,9];
        bool[,] subBox = new bool[9,9];

        for(int i=0; i<9; i++){
            for(int j=0; j<9; j++){
                int value;
                bool converted = int.TryParse((board[i][j]).ToString(), out value);
                if(!converted) continue;
                if(rows[i,value-1]) return false;
                else rows[i,value-1] = true;
                if(cols[j,value-1]) return false;
                else cols[j,value-1]= true;

                int box = -1;
                

                if(0<=i && i<=2 && 0<=j && j<=2) box=0;
                else if(0<=i && i<=2 && 3<=j && j<=5) box=1;
                else if(0<=i && i<=2 && 6<=j && j<=8) box=2;
                else if(3<=i && i<=5 && 0<=j && j<=2) box=3;
                else if(3<=i && i<=5 && 3<=j && j<=5) box=4;
                else if(3<=i && i<=5 && 6<=j && j<=8) box=5;
                else if(6<=i && i<=8 && 0<=j && j<=2) box=6;
                else if(6<=i && i<=8 && 3<=j && j<=5) box=7;
                else if(6<=i && i<=8 && 6<=j && j<=8) box=8;

                if(subBox[box,value-1]) return false;
                else subBox[box,value-1]=true;
            }
        }
        return true;
    }
}

File: valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
 var isValidSudoku = function(board) {
    const N = board.length
    let rows = new Array(N)
    let cols = new Array(N)
    let boxes = new Array(N)

    for (let i = 0; i < N; i++) {
        rows[i] = new Set()
        cols[i] = new Set()
        boxes[i] = new Set()
    }

    for (let i = 0; i < N; i++) {
        for (let j = 0; j < N; j++) {
            let item = board[i][j]
            let box = getBox(i,j)
            if(item == ".") continue
            if(rows[i].has(item)) return false
            if(cols[j].has(item)) return false
            if(boxes[box].has(item)) return false

            rows[i].add(item)
            cols[j].add(item)
            boxes[box].add(item)
        } 
    }
  
    return true
};
var getBox = function(x,y){
    if(0 <= x && x <= 2){
        if(0 <= y && y <= 2) return 0
        else if(3 <= y && y <= 5) return 1
        else return 2
    }
    else if(3 <= x && x <= 5){
        if(0 <= y && y <= 2) return 3
        else if(3 <= y && y <= 5) return 4
        else return 5
    }
    else{
        if(0 <= y && y <= 2) return 6
        else if(3 <= y && y <= 5) return 7
        else return 8
    }
};